proxy_agent/src/data_vessel.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+use crate::key_keeper::key::Key;
+use std::sync::mpsc::Sender;
+
+pub enum DataAction {
+    Stop,
+    KeyKeeperSetKey {
+        key: Key,
+        response: std::sync::mpsc::Sender<bool>,
+    },
+    KeyKeeperGetKeyValue {
+        response: std::sync::mpsc::Sender<Key>,
+    },
+}
+
+#[derive(Clone, Debug)]
+pub struct DataVessel {
+    sender: Sender<DataAction>,
+}
+impl DataVessel {
+    pub fn start_new_async() -> Self {
+        let (sender, receiver) = std::sync::mpsc::channel::<DataAction>();
+
+        std::thread::spawn(move || {
+            // chached data are defined here
+            let mut cached_key: Key = Key::empty();
```

Comment:
The empty() provides the empty/default values.

Reply:
Yes, I understand what it's doing, what I'm saying is it's not the idiomatic way to do it. The proper way to provide a default is by implementing the `Default` trait. However, I'm also suggesting you might not want to use the default approach at all.

This interface is easy to misuse. The default key can't possibly be useful, right? What's the failure mode if the default key is used by whatever requested it? I assume the best case scenario is the that something the requester wants to do fails, but it could also be the case it's used in such a way that using a well-known default is a security concern.

In either case, you should ensure in this API that it _cannot_ be misused if at all possible, and that is easy to do by wrapping the key in an Option and not having a default function.